Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -182,15 +196,35 @@ bool loadCanceled
           errorReporter.ReportBoogieError(outcomeError);
         }
 
-        var diagnostics = errorReporter.GetDiagnostics(document.Uri).OrderBy(d => d.Range.Start).ToList();
-        concurrentDictionary.AddOrUpdate(methodPosition, diagnostics, (_, _) => diagnostics);
+        return errorReporter.GetDiagnostics(document.Uri).OrderBy(d => d.Range.Start).ToList();
+      });
+
+      var result = implementationTasks.Select(it => it.ObservableStatus.Select(async _ => {
+        if (it.CurrentStatus is VerificationStatus.Completed) {
+          var itDiagnostics = await diagnostics[it];
+          var methodPosition = it.Implementation.tok.GetLspPosition();
+
+          concurrentDictionary.AddOrUpdate(methodPosition, itDiagnostics, (_, _) => itDiagnostics);
+        }
+
+        if (it.CurrentStatus is VerificationStatus.Running) {
+          // For backwards compatibility
```

Comment:
Also, this [method](https://github.com/dafny-lang/dafny/pull/1942/files#diff-7f985d6ada10554d2ab71064d97122fa429c78a930bb2d6fb8841d2db74bababR392-R400) redefines these notifications so that it displays "number of methods verified / total number of methods", along with the name of all methods that are being verified.
Will you follow-up with a PR to implement that change?

Reply:
> You can remove this comment. It's not just "backwards compatibility", it's a documented and tested feature.

I'd like to distinguish between the IDE feature for notifications in the status bar and the API that currently supports it. The information in `notificationPublisher.SendStatusNotification(document.TextDocumentItem, CompilationStatus.VerificationStarted, it.Implementation.Name)` is also contained in the `FileVerificationStatus` API, so I think it would be better to stop sending the former after we update the IDE to only use the information from `FileVerificationStatus`.

What do you think?

Reply:
Good point. Although, I can imagine changing the status bar to say 

"Verifying 4 proofs simultaneously (23/70)" when it's verifying more than 1 proof at a time, and say

"Verifying MethodNamedFoo (69/70)" when it's only verifying 1 proof.

That way you are still informed if your verification is hanging on a particular proof, and you're also informed of how concurrent the verification is. Opinions @cpitclaudel ?

Reply:
This scales better with the number of processor than listing all the methods that are being verified, especially in the little space that the status bar offers. I like it well, although I would not be against providing one name at least, e.g. the last method which started to verify.